src/main/scala/org/allenai/ike/IkeBatchAPI.scala

```diff
@@ -0,0 +1,226 @@
+    *       name = "treatements",
+    *       pattern = "{were given|treated with|received|receiving} {CD|NN|JJ|IN}+"
+    *     }
+    *   ]
+    * }
+    *
+    * IndexWithContent = "RCTs"
+    * -------------------------
+    *
+    * @param configFile The config file that holds your patterns.
+    * @return a Map of (String, SearchRequest) where the key (the String) is the name from the config file
+    */
+  def loadPatternsFromConfig(configFile: String): Map[String, SearchRequest] = {
+    val patternConfig = ConfigFactory.load(configFile)
+    val hugeLimit = 100000000
+    val extractionPatterns = patternConfig.getConfigList("ExtractionPatterns.patterns").asScala.map { config =>
+      config.getString("name") -> SearchRequest(Left(config.getString("pattern")), None, None, SearchConfig(hugeLimit, 1))
+    }.toMap
+    extractionPatterns
+  }
+
+  /** Peforms the search over the corpus given the specific pattern to look for (which was loaded from config)
+    *
+    *   // TODO: Should probably refactor to reuse parts of IKEToolWebApp, maybe some common pieces for both?
+    *   // TODO: lots of copypasta from IKEToolWebApp, gross
+    *
+    * @param patternName The pattern you are looking for
+    * @param req The SearchRequest object
+    * @param indexWithContent This is the name of hte index that holds hte content you care about. You need it
+    *                         because you want the filename and the sentence/content for this pattern's match
+    * @return A Seq of ApiExtractionResult case classes which hold the results
+    */
+  def doSearch(patternName: String, req: SearchRequest, indexWithContent: String): Seq[ApiExtractionResult] = {
+    import org.allenai.ike.JsonSerialization._
+
+    val query = SearchApp.parse(req).get
+
+    val (tables, patterns) = req.userEmail match {
+      case Some(userEmail) => (
+        Tablestore.tables(userEmail),
+        Tablestore.namedPatterns(userEmail)
+      )
+      case None => (Map.empty[String, Table], Map.empty[String, NamedPattern])
+    }
+
+    val interpolatedQuery = QueryLanguage.interpolateQuery(
+      query,
+      tables,
+      patterns,
+      word2vecPhrasesSearcher,
+      tableExpander
+    ).get
+
+    val foundResults = searchApps.map { case (name, searcher) => searcher.search(interpolatedQuery, req.config).get }
+    val qexpr = SearchApp.parse(req).get
+
+    // brakes the flatmap if empty
+    if (foundResults.head.isEmpty) {
+      return List[ApiExtractionResult]()
+    }
+
+    val extractions = foundResults.flatMap {
+      case rr =>
+        rr.map {
+          case blab =>
+            val docId = blab.sourceDocument.getOrElse(-1) // TODO: Should barf on -1
+
+            val extractedString = blab.matchWords.mkString(" ")
+
+            val extractionStart = blab.matchOffset.start
+            val extractionEnd = blab.matchOffset.end
+
+            // Lucene document is XML. Assuming one index where this documnent lives for now. TODO: sketchy...
```

Comment:
Yeah, I agree. :) very sketchy...

The issue is that I would like to associate each extraction with the file it came from and the text of the string associated with it (the original string). Do you have any ideas how to do this in a better way? If document IDs are assumed to be unique across the indices, then I can just try to get the document from each index, catch (and ignore) all the errors, and pass along the value from the one that matches, or something. But I think there might be a cleaner way?